src/web/app/pages-student/student-home-page/student-home-page.component.spec.ts

```diff
@@ -36,4 +36,59 @@ describe('StudentHomePageComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with null fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with blank fields', () => {
+    const studentName: string = '';
+    const studentCourse: any = {
+      course: {},
+      feedbackSessions: [],
+    };
+    component.user = studentName;
+    component.courses = [studentCourse];
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with populated fields', () => {
+    const studentName: string = 'John Doe';
+    const studentCourse: any = {
+      course: {
+        id: 'CS2103',
+        name: 'Software Engineering',
+      },
+      feedbackSessions: [
+        {
+          feedbackSession:
+              {
+                feedbackSessionName: 'First Session',
+                courseID: 'CS3281',
+              },
+        },
+        {
+          feedbackSession:
+              {
+                feedbackSessionName: 'Second Session',
+                courseID: 'CS3282',
+              }
+        }
+      ],
+    };
+    const sessionInfoMap: any = {
+      endTime: '1200',
+      isOpened: true,
+      isWaitingToOpen: true,
+      isPublished: true,
+      isSubmitted: true,
+    };
+
+    component.user = studentName;
+    component.courses = [studentCourse];
+    component.sessionsInfoMap.set('Session', sessionInfoMap);
```

Comment:
If you check the HTML file, you'll notice that the key is constructed as `courseId` + `%` + `feedbackSessionName`, e.g. it should be `CS3281%First Session` and `CS3282%Second Session` respectively, in order to test this.

Reply:
Fixed :)